File: api/src/main/kotlin/org/team619/rutor/model/DetailPage.java
package org.team619.rutor.model;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import org.team619.rutor.model.core.DefaultEntity;

import java.util.List;

/**
 * Created by BORIS on 07.08.2016.
 */
@Getter
@Setter
@Builder
public class DetailPage implements DefaultEntity {

    private final String id;
    private final String name;
    private final String fileName;
    private final String link;
    private final String body;
    private final String details;
    private final List<Row> bounded;
    private final List<Comment> comments;

}

File: api/src/main/kotlin/org/team619/rutor/model/DetailPage.kt
package org.team619.rutor.model

import org.team619.rutor.core.Page
import java.util.*

/**
 * Created by BORIS on 07.08.2016.
 */
data class DetailPage(
        val id: String = ""
        , val name: String = ""
        , val fileName: String = ""
        , val link: String = ""
        , val body: String = ""
        , val details: String = ""
        , val bounded: List<Row> = ArrayList()
        , val comments: List<Comment> = ArrayList()) : Page {

    override fun toString(): String {
        return "DetailPage(id='$id', name='$name', fileName='$fileName', link='$link', details='$details', bounded=$bounded, comments=$comments)"
    }

}
